heron/instance/src/java/com/twitter/heron/instance/Slave.java

```diff
@@ -62,6 +74,13 @@ public Slave(SlaveLooper slaveLooper,
     this.streamInCommunicator = streamInCommunicator;
     this.streamOutCommunicator = streamOutCommunicator;
     this.inControlQueue = inControlQueue;
+    this.metricsOutCommunicator = metricsOutCommunicator;
+    this.serializer =
+        SerializeDeSerializeHelper.getSerializer(helper.getTopologyContext().getTopologyConfig());
+
+    // TODO (nlu): Create the state with corresponding type according to config
+    instanceState = new HashMapState<>();
```

Comment:
When `HeronInstance` starts the slave thread, it also doesn't know what's the state for it. So it's not the guy who can inject the state into an instance. Actually, stream manager is the one who sends the state to a local instance via the network. Instead of an empty state, a `null` might be more proper here.